source/common/router/router.cc

```diff
@@ -704,8 +704,31 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   // Hang onto the modify_headers function for later use in handling upstream responses.
   modify_headers_ = modify_headers;
 
-  UpstreamRequestPtr upstream_request =
-      std::make_unique<UpstreamRequest>(*this, std::move(generic_conn_pool));
+  bool as_early_data{false};
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.conn_pool_new_stream_with_early_data_and_alt_svc")) {
+    absl::string_view method = downstream_headers_->getMethodValue();
+    const bool is_safe_method = (method == Http::Headers::get().MethodValues.Get ||
```

Comment:
Might consider referencing the RFC definition of safe in a comment here.

Reply:
done